hazelcast/src/main/java/com/hazelcast/collection/impl/queue/QueueContainer.java

```diff
@@ -64,8 +64,8 @@
     private static final int ID_PROMOTION_OFFSET = 100000;
     private final Map<Long, TxQueueItem> txMap = new HashMap<Long, TxQueueItem>();
     private final Map<Long, Data> dataMap = new HashMap<Long, Data>();
-    private final QueueWaitNotifyKey pollWaitNotifyKey;
-    private final QueueWaitNotifyKey offerWaitNotifyKey;
+    private QueueWaitNotifyKey pollWaitNotifyKey;
```

Comment:
It's not possible to keep those fields as `final`, `name` is not yet known and we wouldn't be able to create those wait/notify keys at deserialization time in `readData`, based on the `name` that is deserialized.